### PR TITLE
PACA-486: convert code to use rda schema

### PR DESCRIPTION
**Pull request by brianburton**
**JIRA Ticket:**
[PACA-486](https://jira.cms.gov/browse/PACA-486)

**User Story or Bug Summary:**

Change Application code to use new RDA Schema

---

### What Does This PR Do?

- Modifies all RDA JPA entity classes to reference the new `rda` schema and tables instead of the old `pre_adj` versions.
- Renames all RDA JPA entity classes to use prefix `Rda` for their names instead of `PreAdj`.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* The version of master containing the merged [PR for PACA-328](https://github.com/CMSgov/beneficiary-fhir-data/pull/1022) must be deployed before this PR can be safely merged.

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by KyRobbins — APPROVED** [review]

**Review by ericbuckley — APPROVED** [review]

**Review by ericbuckley — CHANGES_REQUESTED** [review]
BLOCKED from merge until #1018 is deployed to all AWS environments

**Review by ericbuckley — APPROVED** [review]

File: apps/bfd-pipeline/bfd-pipeline-shared-utils/src/main/java/gov/cms/bfd/pipeline/sharedutils/PipelineTestUtils.java [hunk 1]
```diff
@@ -212,6 +203,25 @@ public void truncateTablesInDataSource() {
     }
   }
 
+  /**
+   * Schemas that use mixed case use quotes and have their original case preserved but those without
+   * quotes are converted to upper case to be compatible with Hibernate/HSQLDB.
+   *
+   * <p>Note: This may need to be quoted on PostgreSQL. If so, since HSQL DB blows up if we quote
+   * them, this code may have to first check the DB platform. TBD.
```
Comment (christrotteradhoc) [inline apps/bfd-pipeline/bfd-pipeline-shared-utils/src/main/java/gov/cms/bfd/pipeline/sharedutils/PipelineTestUtils.java:211]: can we determine this before going forward by running against postgres locally?

**Review by mackec1 — APPROVED** [review]
It looks good (especially getting rid of the back-ticks!); the one nit (and I'm sure trotter will weigh in on this as well) is the paucity of comments and/or Javadoc. I believe Logan is working on a combo linter (no problem in your code) as well as Javadoc enforecement.

**Review by ericbuckley — APPROVED** [review]

File: apps/bfd-pipeline/bfd-pipeline-shared-utils/src/main/java/gov/cms/bfd/pipeline/sharedutils/PipelineTestUtils.java (hunk 1 above)
Comment (brianburton) [reply apps/bfd-pipeline/bfd-pipeline-shared-utils/src/main/java/gov/cms/bfd/pipeline/sharedutils/PipelineTestUtils.java]: Thanks.  I tested with postgres and found problems.  These tests run correctly in HSQLDB and Postgresql now.

**Comment by lsmitchell** [comment]
> It looks good (especially getting rid of the back-ticks!); the one nit (and I'm sure trotter will weigh in on this as well) is the paucity of comments and/or Javadoc. I believe Logan is working on a combo linter (no problem in your code) as well as Javadoc enforecement.

The checkstyle enforcer is checked in now, but only for one project (shared-utils). While the new doc enforcement still applies (via PR nagging) for new code in all projects, most of this is adjustments AND we agreed to make RDA classes lower priority since they'd be likely generated in the near future.

File: apps/bfd-pipeline/bfd-pipeline-shared-utils/src/main/java/gov/cms/bfd/pipeline/sharedutils/PipelineTestUtils.java [hunk 2]
```diff
@@ -212,6 +204,50 @@ public void truncateTablesInDataSource() {
...
+      tableNameSpecifier = tableNameSpecifier.replaceAll("`", "");
+    } else {
+      tableNameSpecifier = tableNameSpecifier.toUpperCase();
+    }
+    return tableNameSpecifier;
+  }
+
+  /**
+   * We're stuck in a painful situation where postgresql only recognizes the schema in its original
+   * case but HSQLDB only recognizes the schema in all uppercase.
```
Comment (lsmitchell) [inline apps/bfd-pipeline/bfd-pipeline-shared-utils/src/main/java/gov/cms/bfd/pipeline/sharedutils/PipelineTestUtils.java:241]: Ran into the same situation during the migration testing too.

**Review by lsmitchell — APPROVED** [review]

**Review by ericbuckley — APPROVED** [review]

**Pull request merged**